lib/calendars.ml

```diff
@@ -4,10 +4,29 @@
 (* Changed gregorian and julian to work always with negative years
    (Scott's version worked only for years > -4800 *)
 
-type d = { day : int ; month : int ; year : int ; delta : int }
+type gregorian
+type julian
+type french
+type hebrew
+
+type _ kind =
+  | Gregorian : gregorian kind
+  | Julian : julian kind
+  | French : french kind
+  | Hebrew : hebrew kind
+
+type 'a date = {
+  day : int;
+  month : int;
+  year : int;
+  delta : int;
+  kind : 'a kind;
+}
+
+type sdn = int
 
 let mydiv x y = if x >= 0 then x / y else (x - y + 1) / y
-let mymod x y = if x >= 0 then x mod y else (x + 1) mod y + y - 1
+let mymod x y = if x >= 0 then x mod y else ((x + 1) mod y) + y - 1
```

Comment:
Maybe writing something like:

```ocaml
let ( // ) x y = if ...
let ( %% ) x y = if ...
```

Would help to make the following code more readable ?